### PR TITLE
Provide shorter and nicer api for setting up a service transport #404

### PR DESCRIPTION
**Pull request by snripa**

### CODE REVIEW

File: services/src/main/java/io/scalecube/services/Microservices.java [hunk 1]
```diff
@@ -253,23 +295,8 @@ public Builder discovery(Consumer<ServiceDiscoveryConfig.Builder> discoveryOptio
...
-    }
-
-    public Builder servicePort(int port) {
-      this.transportBootstrap.servicePort(port);
-      return this;
-    }
-
-    public Builder numOfThreads(int numOfThreads) {
-      this.transportBootstrap.numOfThreads(numOfThreads);
+    public Builder transportConfig(ServiceTransportConfig serviceTransportOptions) {
```
Comment (segabriel) [inline services/src/main/java/io/scalecube/services/Microservices.java]: it should look like the following
```java
public Builder transport(Consumer<ServiceTransportConfig.Builder> transportOptions) {
   ...
}
```

File: services/src/main/java/io/scalecube/services/transport/ServiceTransportConfig.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import io.scalecube.services.transport.api.ServiceTransport;
+import io.scalecube.services.transport.api.WorkerThreadChooser;
+
+public class ServiceTransportConfig {
+  private Integer servicePort;
+  private String serviceHost;
+  private ServiceTransport transport;
+  private Integer numOfThreads;
+  private WorkerThreadChooser workerThreadChooser;
+
```
Comment (artem-v) [inline services/src/main/java/io/scalecube/services/transport/ServiceTransportConfig.java:13]: Fields are missing `final` qualifier

File: services/src/main/java/io/scalecube/services/Microservices.java (hunk 1 above)
Comment (artem-v) [reply services/src/main/java/io/scalecube/services/Microservices.java]: 100%

**Review by artem-v — APPROVED** [review]

**Review by artem-v — DISMISSED** [review]
Pressed _Approve_ by mistake last time

**Review by dmytro-lazebnyi — APPROVED** [review]

**Pull request merged**